src/geogenalg/selection.py

```diff
@@ -0,0 +1,139 @@
+
+from collections import defaultdict
+
+import geopandas as gpd
+import pandas as pd
+from shapely.geometry import LineString, MultiLineString, Point
+
+from geogenalg.continuity import find_all_endpoints
+
+
+def remove_disconnected_short_lines(
+    input_gdf: gpd.GeoDataFrame, length_threshold: float
+) -> gpd.GeoDataFrame:
+    """Remove short lines from a GeoDataFrame unless they are connected to other lines.
+
+    A line is retained if:
+    - Its length is greater than or equal to the length_threshold, OR
+    - It is connected to at least two other lines via shared endpoints.
+
+    Args:
+    ----
+        input_gdf: A GeoDataFrame containing LineString geometries
+        length_threshold: The minimum length a line must have to be kept, unless it's
+            well-connected.
+
+    Returns:
+    -------
+        A filtered GeoDataFrame containing only the lines that meet the length or
+            connectivity criteria.
+
+    """
+    lines = list(input_gdf.geometry)
+    endpoints = find_all_endpoints(lines)
+    min_connections = 2
+
+    # Build mapping: Point -> set of line indices touching this point
+    point_to_lines: defaultdict[Point, set[int]] = defaultdict(set)
+    for point, idx, _ in endpoints:
+        point_to_lines[point].add(idx)
+
+    # Count how many endpoints of each line are shared with other lines
+    connected_ends: defaultdict[int, int] = defaultdict(int)
+    for indices in point_to_lines.values():
+        if len(indices) > 1:
+            for idx in indices:
+                connected_ends[idx] += 1
+
+    # Determine which lines to keep
+    lines_to_keep = []
+    for idx, geom in enumerate(lines):
+        line_length = geom.length
+        connection_count = connected_ends[idx]
+        keep = (line_length >= length_threshold) or (
+            connection_count >= min_connections
+        )
+        lines_to_keep.append(keep)
+
+    return input_gdf[lines_to_keep]
+
+
+def remove_large_polygons_containing_point(
+    polygon_gdf: gpd.GeoDataFrame,
+    point_gdf: gpd.GeoDataFrame,
+    area_threshold_without_point: float,
+    area_threshold_with_point: float,
+) -> gpd.GeoDataFrame:
+    """Remove polygons based on area thresholds and whether they contain any points.
+
+    Args:
+    ----
+        polygon_gdf: A GeoDataFrame containing polygon geometries
+        point_gdf: A GeoDataFrame containing point geometries
+        area_threshold_without_point: Maximum allowed area for polygons that do not
+            contain a point
+        area_threshold_with_point: Maximum allowed area for polygons that do contain
+            a point
+
+    Returns:
+    -------
+        A filtered GeoDataFrame containing only the polygons that meet the criteria.
+
+    """
+    polygons_intersecting_point = polygon_gdf.geometry.apply(
+        lambda poly: point_gdf.geometry.intersects(poly).any()
+    )
+
+    with_point = polygon_gdf[polygons_intersecting_point]
```

Comment:
It seems that VS Code doesn't recognise "geometry" correctly as a Geoseries in the following lines and it's methods/propertys are not typed. Therefore you could add typings to these two earlier lines so that types and editing work nicely:

    with_point: gpd.GeoDataFrame = ...
    without_point: gpd.GeoDataFrame = ...